hedera-mirror-rosetta/app/services/construction/construction_service_test.go

```diff
@@ -0,0 +1,457 @@
+}
+
+func TestConstructionCombine(t *testing.T) {
+	// given:
+	expectedConstructionCombineResponse := &types.ConstructionCombineResponse{
+		SignedTransaction: validTxHash,
+	}
+
+	// when:
+	res, e := NewConstructionAPIService().ConstructionCombine(nil, dummyConstructionCombineRequest())
+
+	// then:
+	assert.Equal(t, expectedConstructionCombineResponse, res)
+	assert.Nil(t, e)
+}
+
+func TestConstructionCombineThrowsWhenNoSingleSignature(t *testing.T) {
+	// given:
+	exampleConstructionCombineRequest := dummyConstructionCombineRequest()
+	exampleConstructionCombineRequest.Signatures = []*types.Signature{}
+
+	// when:
+	res, e := NewConstructionAPIService().ConstructionCombine(nil, exampleConstructionCombineRequest)
+
+	// then:
+	assert.Nil(t, res)
+	assert.Equal(t, errors.Errors[errors.MultipleSignaturesPresent], e)
+}
+
+func TestConstructionCombineThrowsWhenDecodeStringFails(t *testing.T) {
+	// given:
+	exampleInvalidTxHashConstructionCombineRequest := dummyConstructionCombineRequest()
+	exampleInvalidTxHashConstructionCombineRequest.UnsignedTransaction = invalidTxHash
+
+	// when:
+	res, e := NewConstructionAPIService().ConstructionCombine(nil, exampleInvalidTxHashConstructionCombineRequest)
+
+	// then:
+	assert.Nil(t, res)
+	assert.Equal(t, errors.Errors[errors.TransactionDecodeFailed], e)
+}
+
+func TestConstructionCombineThrowsWithInvalidPublicKey(t *testing.T) {
+	// given:
+	exampleInvalidPublicKeyConstructionCombineRequest := dummyConstructionCombineRequest()
+	exampleInvalidPublicKeyConstructionCombineRequest.Signatures[0].PublicKey = &types.PublicKey{}
+
+	// when:
+	res, e := NewConstructionAPIService().ConstructionCombine(nil, exampleInvalidPublicKeyConstructionCombineRequest)
+
+	// then:
+	assert.Nil(t, res)
+	assert.Equal(t, errors.Errors[errors.InvalidPublicKey], e)
+}
+
+func TestConstructionCombineThrowsWhenSignatureIsNotVerified(t *testing.T) {
+	// given:
+	exampleInvalidSigningPayloadConstructionCombineRequest := dummyConstructionCombineRequest()
+	exampleInvalidSigningPayloadConstructionCombineRequest.Signatures[0].SigningPayload = &types.SigningPayload{}
+
+	// when:
+	res, e := NewConstructionAPIService().ConstructionCombine(nil, exampleInvalidSigningPayloadConstructionCombineRequest)
+
+	// then:
+	assert.Nil(t, res)
+	assert.Equal(t, errors.Errors[errors.InvalidSignatureVerification], e)
+}
+
+func TestConstructionDerive(t *testing.T) {
+	// when:
+	res, e := NewConstructionAPIService().ConstructionDerive(nil, nil)
+
+	// then:
+	assert.Nil(t, res)
+	assert.Equal(t, errors.Errors[errors.NotImplemented], e)
+}
+
+func TestConstructionHash(t *testing.T) {
+	// given:
+	exampleConstructionHashRequest := dummyConstructionHashRequest(validTxHash)
+	exampleConstructionHashResponse := &types.TransactionIdentifierResponse{
+		TransactionIdentifier: &types.TransactionIdentifier{Hash: "0x9768d458c755befcda5c6fca07e9f7693b94c429f9c414b0cea07163c402ddd44d1108f89b190d0dcabc423a3d45696d"},
```

Comment:
Wouldn't it be better if we define this hash at the top of the file as a variable similar to the others

Reply:
My reasoning behind this decision was that this hash is only used in this function specifically. IMO I'd rather put it in a variable on at the top of the current function.

Reply:
Okay, you are right. In this case, can we introduce it in a variable in the scope of the test and give a meaningful name to the hash. That is my main reason for wanting it to be in a variable because the variable will signal that kind of transaction is this.